lib/data/http/apis/inventory_api.dart

```
@@ -28,15 +29,37 @@ class InventoryApi extends HttpApi<InventoryService> {
     );
   }
 
-  Future<void> accountWalletTransactions(
+  Future<List<AssetTransaction>> accountWalletTransactions(
     AccountWallet accountWallet, {
+    String? transactionId,
     required String currency,
-    required DateTime fromDate,
-    required DateTime toDate,
-    required int page,
     required int pageSize,
-  }) {
-    throw UnimplementedError();
+    bool loadLatest = false,
+  }) async {
+    final res = await service.accountWalletTransactions(
+      accountWallet.toAccountTransactionsRequest(
+        currency: currency,
+        transactionId: transactionId,
+        pageSize: pageSize,
+        latest: loadLatest,
+      ),
+    );
+
+    final error = res.error;
+    if (error is HttpError) throw error;
+
```

Here is if (error is HttpError) throw error; But little lower you have throw const UnknownHttpError. Doesn't the first if already catch it? I think we should use another type of error or?

---

yes. it is due to our error convertor that is provided to http client. if there any error that is recognised inside error convertor we catch it but this line `if (error is HttpError)` and we let it bubble up. However in case the response body is `null` and the error was not recognised we throw UnknownError.